### PR TITLE
fix: Improve boot time query on Windows

### LINKED ISSUE #955: Azure CNI creates lots of short lived calls to WMI
**What happened:** 
Azure CNI creates lots of short lived calls to WMI.  

1. Wmic.exe processes were launched by azure-vnet.exe in session 0 to query system bootup time.
2. azure-vnet.exe is short lived process and the calls to WMI are expensive

what is the purpose and is there away to optimize these calls?

image

**What you expected to happen:**  

**How to reproduce it:**  

**Orchestrator and Version (e.g. Kubernetes, Docker):**  

**Operating System (Linux/Windows):**  

**Kernel (e.g. `uanme -a` for Linux or `$(Get-ItemProperty -Path "C:\windows\system32\hal.dll").VersionInfo.FileVersion` for Windows):**

**Anything else we need to know?:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by zhiweiv**
**Reason for Change**:
Improve performance of boot time query on Windows, it is also a solution for https://github.com/Azure/azure-container-networking/issues/955.

This is refresh of https://github.com/Azure/azure-container-networking/pull/529, the idea comes from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/util_windows.go#L119.

Please note there will be possible one second deviation of result in new implementation, for example the output:
Old:   2021-08-09 00:52:28
New: 2021-08-09 00:52:27(the actual value is 2021-08-09 00:52:27.3958047)

**Issue Fixed**:
Fixes #955

**Requirements**:
- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests

### CODE REVIEW

File: platform/os_windows.go [hunk 1]
```diff
@@ -72,40 +72,18 @@ func GetProcessSupport() error {
...
-	bootHour := systemBootupTime[8:10]
-	bootMin := systemBootupTime[10:12]
-	bootSec := systemBootupTime[12:14]
-	systemBootTime := bootYear + "-" + bootMonth + "-" + bootDay + " " + bootHour + ":" + bootMin + ":" + bootSec
-
-	log.Printf("Formatted Boot time: %s", systemBootTime)
-
-	// Parse the boot time.
-	layout := "2006-01-02 15:04:05"
-	rebootTime, err := time.ParseInLocation(layout, systemBootTime, time.Local)
```
Comment (jsturtevant) [inline platform/os_windows.go:103]: Will this cause a difference in output in different regions?

Comment (tamilmani1989) [reply platform/os_windows.go:103]: based on timezone, its printing the reboot time. 

pst:

2021/08/09 13:19:34 [12816] Formatted Boot time: 2021-07-20 15:02:32

i then set it to ist:

2021/08/10 01:49:58 [17720] Formatted Boot time: 2021-07-21 03:32:32

File: platform/os_windows.go [hunk 2]
```diff
@@ -72,40 +72,18 @@ func GetProcessSupport() error {
...
-		log.Printf("Failed to parse boot time, err:%v", err)
+	currentTime := time.Now()
+	output, _, err := tickCount.Call()
+	if errno, ok := err.(syscall.Errno); !ok || errno != 0 {
+		log.Printf("Failed to call GetTickCount64, err: %v", err)
 		return time.Time{}.UTC(), err
 	}
-
+	rebootTime := currentTime.Add(-time.Duration(output) * time.Millisecond).Truncate(time.Second)
+	log.Printf("Formatted Boot time: %s", rebootTime.Format("2006-01-02 15:04:05"))
```
Comment (tamilmani1989) [inline platform/os_windows.go:86]: how about this? This has tz info too. => `Formatted Boot time: 2021-07-20T15:02:33-07:00`
```suggestion
	log.Printf("Formatted Boot time: %s", rebootTime.Format(time.RFC3339))
```

Comment (zhiweiv) [reply platform/os_windows.go:86]: So, should I commit this change?

Comment (tamilmani1989) [reply platform/os_windows.go:86]: yes please. we have a release tomorrow. can you please address this asap?

Comment (zhiweiv) [reply platform/os_windows.go]: Change committed.

**Comment by zhiweiv** [comment]
Just a question, I saw a e2e job (E2E - Ubuntu 16.04 Singletenancy AKS Engine Suite - (ubuntu_16_04_linux_e2e)), Ubuntu 16.04 support is dropped in AKS Engine, see https://github.com/Azure/aks-engine/pull/4418.

**Comment by tamilmani1989** [comment]
> Just a question, I saw a e2e job (E2E - Ubuntu 16.04 Singletenancy AKS Engine Suite - (ubuntu_16_04_linux_e2e)), Ubuntu 16.04 support is dropped in AKS Engine, see [Azure/aks-engine#4418](https://github.com/Azure/aks-engine/pull/4418).

ubuntu16.04 is not deprecated still. They just recommend to use ubuntu18 since canonical retire support for ubuntu16

`Note that this PR doesn't actually deprecate Ubuntu 16.04-LTS. Its intention is to make it much less likely that someone would build a new cluster backed by 16.04-LTS (the default has been 18.04-LTS for a long time, so any new 16.04-LTS clusters would have to be explicitly configured as such).`

**Review by tamilmani1989 — APPROVED** [review]

**Pull request merged**